pkg/manifests/manifests_test.go

```diff
@@ -779,26 +780,29 @@ func TestPrometheusOperatorConfiguration(t *testing.T) {
 		t.Fatalf("expected node selector to be master, got %q", got)
 	}
 
-	expectedPromOpImage := "docker.io/openshift/origin-prometheus-operator:latest"
-	resPromOpImage := d.Spec.Template.Spec.Containers[0].Image
-	if resPromOpImage != expectedPromOpImage {
-		t.Fatalf("Configuring the Prometheus Operator image failed, expected: %v, got %v", expectedPromOpImage, resPromOpImage)
-	}
-
 	prometheusReloaderFound := false
 	prometheusWebTLSCipherSuitesArg := ""
 	prometheusWebTLSVersionArg := ""
-	for i := range d.Spec.Template.Spec.Containers[0].Args {
-		if strings.HasPrefix(d.Spec.Template.Spec.Containers[0].Args[i], PrometheusConfigReloaderFlag+"docker.io/openshift/origin-prometheus-config-reloader:latest") {
-			prometheusReloaderFound = true
-		}
-
-		if strings.HasPrefix(d.Spec.Template.Spec.Containers[0].Args[i], PrometheusOperatorWebTLSCipherSuitesFlag) {
-			prometheusWebTLSCipherSuitesArg = d.Spec.Template.Spec.Containers[0].Args[i]
-		}
+	kubeRbacProxyTLSCipherSuitesArg := ""
+	kubeRbacProxyMinTLSVersionArg := ""
+	for _, container := range d.Spec.Template.Spec.Containers {
+		switch container.Name {
+		case "prometheus-operator":
+			if container.Image != "docker.io/openshift/origin-prometheus-operator:latest" {
+				t.Fatalf("%s image incorrectly configured", container.Name)
+			}
 
-		if strings.HasPrefix(d.Spec.Template.Spec.Containers[0].Args[i], PrometheusOperatorWebTLSMinTLSVersionFlag) {
-			prometheusWebTLSVersionArg = d.Spec.Template.Spec.Containers[0].Args[i]
+			if expectContainerArg(d.Spec.Template.Spec.Containers, PrometheusConfigReloaderFlag+"docker.io/openshift/origin-prometheus-config-reloader:latest") != "" {
+				prometheusReloaderFound = true
+			}
+			prometheusWebTLSCipherSuitesArg = expectContainerArg(d.Spec.Template.Spec.Containers, PrometheusOperatorWebTLSCipherSuitesFlag)
```

Comment:
Thanks for the change @slashpai, this looks much better now. I am just wondering if we should call the function something more descriptive. Is the function supposed to make an assertion, or return the value for the field? If it's the former then it should be `assertContainerArgEquals`, otherwise it should be something like `getContainerArgValue`

Reply:
Same comment. I think with the current naming I would just expect the test to fail if the assertion was not met. Renaming to `getContainerArgValue` seems more appropriate.

Also, looking at the code, we seem to already range over the containers so perhaps `getContainerArgValue` does not need to take a slice

Reply:
I updated the func name to `getContainerArgValue()`
`d.Spec.Template.Spec.Containers` is of type of `[]v1.Container` so that should be the type right?

Reply:
@fpetkovski @PhilipGough Please review :)